# **Build-a-Planet**

Division of Project:
* Code Lead: Rachel Price
* Report Lead: Ethan Johnson
* Presentation Lead: Malachi Roark

Throughout this notebook, we will use ExoPlex to calculate a planet’s possible density and structure. Calculate planet structures from mass-proportions and compare to known stellar values. We will then place the structure and composition of our planet in context of mineral proportions and orbital parameters.

## **Importing ExoPlex**

In [1]:
from google.colab import drive
drive.mount('/content/drive', force_remount=True)

Mounted at /content/drive


In [2]:
cd /content/drive/MyDrive/ExoPlex_dist/

/content/drive/.shortcut-targets-by-id/1CklpevfgBa21F5mnzM59y_7t9YrIFMED/ExoPlex_dist


In [3]:
cd Group_7/

/content/drive/.shortcut-targets-by-id/1CklpevfgBa21F5mnzM59y_7t9YrIFMED/ExoPlex_dist/Group_7


In [4]:
!ls -lrt

total 3561
-rw------- 1 root root   6573 Mar 17  2021  single_phase.py
-rw------- 1 root root   9053 Mar 17  2021  MR_Trappist.py
-rw------- 1 root root   6888 Mar 17  2021  example_by_radius.py
-rw------- 1 root root   6476 Mar 17  2021  example_by_mass.py
-rw------- 1 root root    500 Mar 17  2021  build.log
-rw------- 1 root root      0 Apr  9  2022  1,1_0,8_0,07_0,09_0,01_0,0_MANTLE.dat
-rw------- 1 root root  64852 Mar 16 20:06 'CRF(1).png'
-rw------- 1 root root   7533 Mar 16 20:55  Group_X.py
-rw------- 1 root root 267846 Mar 29 00:56  high_wFEO_Radius_0.97.tsv
-rw------- 1 root root 267720 Mar 29 00:58  high_wFEO_Radius_1.18.tsv
-rw------- 1 root root 266547 Apr  1 19:29  high_wFEO_Radius_0.52.tsv
-rw------- 1 root root 266261 Apr  1 19:29  high_wFEO_Radius_0.42.tsv
-rw------- 1 root root 266853 Apr  1 19:29  high_wFEO_Radius_1.08.tsv
-rw------- 1 root root 265431 Apr  1 19:29  high_wFEO_Radius_0.79.tsv
-rw------- 1 root root 267431 Apr  1 19:30  high_wFEO_Radius_1.01.tsv
-rw--

## **Warm Up**

### **Question 1**

Calculate the radius of the Earth, Mars, and Mercury given the mass of each planet and using the Sun’s proportions of Fe, Mg, and Si. Compare these values with the actual planets' values.

ExoPlex uses Solar compositions as its default (Defaults: Ca/Mg = 0.07; Si/Mg=0.9; Al/Mg = 0.09; Fe/Mg = 0.9).

**Calculating Earth Radius**

In [5]:
!python Group_X.py --mass=1  # Earth


Core composition:  {'Fe': np.float64(100.0), 'Si': np.float64(0.0), 'O': np.float64(0.0), 'S': np.float64(0.0)}
Mantle composition:  {'FeO': np.float64(0.0), 'SiO2': np.float64(52.55497015), 'MgO': np.float64(39.17101638), 'CaO': np.float64(3.81493827), 'Al2O3': np.float64(4.45907521)}
Mantle Fe# -0.0
Core Mass Percent =  32.816

Your closest grid filename is:  0.07CaMg_0.00FeMg_0.09AlMg_0.9SiMg_0.0NaMg_0.00Fe_U/LM_results.txt


Core composition:  {'Fe': np.float64(100.0), 'Si': np.float64(0.0), 'O': np.float64(0.0), 'S': np.float64(0.0)}
Mantle composition:  {'FeO': np.float64(0.0), 'SiO2': np.float64(52.55497015), 'MgO': np.float64(39.17101638), 'CaO': np.float64(3.81493827), 'Al2O3': np.float64(4.45907521)}
Mantle Fe# -0.0
Core Mass Percent =  32.816

The Upper mantle .tab already exists, please wait briefly for solution

../Solutions_Small/0.07CaMg_0.00FeMg_0.09AlMg_0.9SiMg_0.0NaMg_0.00Fe
The Lower mantle .tab already exists, please wait briefly for solution

iteration # 1
iterati

**Calculating Mars Radius**

In [6]:
import astropy.constants as const
import astropy.units as u

# Calculate mass of Mars in M_earth
mars_mass_earth = (6.4171e23*u.kg) / const.M_earth
print(f'Mass of Mars:',mars_mass_earth,'M_earth')

# Calculate radius of Mars in R_earth
mars_rad_earth = (3.3895e6*u.m) / const.R_earth
print(f'Radius of Mars:',round(mars_rad_earth,3),'R_earth')

Mass of Mars: 0.10745009420462197 M_earth
Radius of Mars: 0.531 R_earth


In [7]:
!python Group_X.py --mass=0.10745009420462197  # Mars


Core composition:  {'Fe': np.float64(100.0), 'Si': np.float64(0.0), 'O': np.float64(0.0), 'S': np.float64(0.0)}
Mantle composition:  {'FeO': np.float64(0.0), 'SiO2': np.float64(52.55497015), 'MgO': np.float64(39.17101638), 'CaO': np.float64(3.81493827), 'Al2O3': np.float64(4.45907521)}
Mantle Fe# -0.0
Core Mass Percent =  32.816

Your closest grid filename is:  0.07CaMg_0.00FeMg_0.09AlMg_0.9SiMg_0.0NaMg_0.00Fe_U/LM_results.txt


Core composition:  {'Fe': np.float64(100.0), 'Si': np.float64(0.0), 'O': np.float64(0.0), 'S': np.float64(0.0)}
Mantle composition:  {'FeO': np.float64(0.0), 'SiO2': np.float64(52.55497015), 'MgO': np.float64(39.17101638), 'CaO': np.float64(3.81493827), 'Al2O3': np.float64(4.45907521)}
Mantle Fe# -0.0
Core Mass Percent =  32.816

The Upper mantle .tab already exists, please wait briefly for solution

../Solutions_Small/0.07CaMg_0.00FeMg_0.09AlMg_0.9SiMg_0.0NaMg_0.00Fe
The Lower mantle .tab already exists, please wait briefly for solution

iteration # 1
iterati

**Calculating Mercury Radius**

In [8]:
# Calculate mass of Mercury in M_earth
mercury_mass_earth = (3.30104e23*u.kg) / const.M_earth
print(f'Mass of Mercury:',mercury_mass_earth,'M_earth')

# Calculate radius of Mercury in R_earth
mercury_rad_earth = (2.4397e6*u.m) / const.R_earth
print(f'Radius of Mercury:',round(mercury_rad_earth,3),'R_earth')

Mass of Mercury: 0.055273730964645294 M_earth
Radius of Mercury: 0.383 R_earth


In [9]:
!python Group_X.py --mass=0.055273730964645294  # Mercury


Core composition:  {'Fe': np.float64(100.0), 'Si': np.float64(0.0), 'O': np.float64(0.0), 'S': np.float64(0.0)}
Mantle composition:  {'FeO': np.float64(0.0), 'SiO2': np.float64(52.55497015), 'MgO': np.float64(39.17101638), 'CaO': np.float64(3.81493827), 'Al2O3': np.float64(4.45907521)}
Mantle Fe# -0.0
Core Mass Percent =  32.816

Your closest grid filename is:  0.07CaMg_0.00FeMg_0.09AlMg_0.9SiMg_0.0NaMg_0.00Fe_U/LM_results.txt


Core composition:  {'Fe': np.float64(100.0), 'Si': np.float64(0.0), 'O': np.float64(0.0), 'S': np.float64(0.0)}
Mantle composition:  {'FeO': np.float64(0.0), 'SiO2': np.float64(52.55497015), 'MgO': np.float64(39.17101638), 'CaO': np.float64(3.81493827), 'Al2O3': np.float64(4.45907521)}
Mantle Fe# -0.0
Core Mass Percent =  32.816

The Upper mantle .tab already exists, please wait briefly for solution

../Solutions_Small/0.07CaMg_0.00FeMg_0.09AlMg_0.9SiMg_0.0NaMg_0.00Fe
The Lower mantle .tab already exists, please wait briefly for solution

iteration # 1
iterati

**Comparing Results to True Values**


* *Estimated Earth Radius*: 0.999 Earth radii
* *True Earth Radius*: 1.0 Earth radii


* *Estimated Mars Radius*: 0.520 Earth radii
* *True Mars Radius*: 0.531 Earth radii


* *Estimated Mercury Radius*: 0.422 Earth radii
* *True Mercury Radius*: 0.383 Earth radii

The estimated Earth radius is nearly identical to the true Earth radius (0.999 versus 1.0 Earth radii). The estimated and true Mars radii are nearly identical as well (0.520 versus 0.531 Earth radii). However, the estimated and true Mercury radii are not as similar (0.422 versus 0.383 Earth radii). This could be due to the fact that Mercury is a dense, rocky planet...significantly more dense than Mars, despite the fact that their masses are quite similar. This explains why the program (not designed to take into account the density of planets, only their mass and the composition of their star) did not estimate the radius of Mercury as accurately.

### **Question 2**

Explore what happens to the size of the planet and the planet structure when you adjust the amount of iron relative to Mg.

In [10]:
!python Group_X.py  # no adjustment (control)


Core composition:  {'Fe': np.float64(100.0), 'Si': np.float64(0.0), 'O': np.float64(0.0), 'S': np.float64(0.0)}
Mantle composition:  {'FeO': np.float64(0.0), 'SiO2': np.float64(52.55497015), 'MgO': np.float64(39.17101638), 'CaO': np.float64(3.81493827), 'Al2O3': np.float64(4.45907521)}
Mantle Fe# -0.0
Core Mass Percent =  32.816

Your closest grid filename is:  0.07CaMg_0.00FeMg_0.09AlMg_0.9SiMg_0.0NaMg_0.00Fe_U/LM_results.txt


Core composition:  {'Fe': np.float64(100.0), 'Si': np.float64(0.0), 'O': np.float64(0.0), 'S': np.float64(0.0)}
Mantle composition:  {'FeO': np.float64(0.0), 'SiO2': np.float64(52.55497015), 'MgO': np.float64(39.17101638), 'CaO': np.float64(3.81493827), 'Al2O3': np.float64(4.45907521)}
Mantle Fe# -0.0
Core Mass Percent =  32.816

The Upper mantle .tab already exists, please wait briefly for solution

../Solutions_Small/0.07CaMg_0.00FeMg_0.09AlMg_0.9SiMg_0.0NaMg_0.00Fe
The Lower mantle .tab already exists, please wait briefly for solution

iteration # 1
iterati

In [11]:
!python Group_X.py --FeMg=1  # equal quantities


Core composition:  {'Fe': np.float64(100.0), 'Si': np.float64(0.0), 'O': np.float64(0.0), 'S': np.float64(0.0)}
Mantle composition:  {'FeO': np.float64(0.0), 'SiO2': np.float64(52.55497015), 'MgO': np.float64(39.17101638), 'CaO': np.float64(3.81493827), 'Al2O3': np.float64(4.45907521)}
Mantle Fe# -0.0
Core Mass Percent =  35.180

Your closest grid filename is:  0.07CaMg_0.00FeMg_0.09AlMg_0.9SiMg_0.0NaMg_0.00Fe_U/LM_results.txt


Core composition:  {'Fe': np.float64(100.0), 'Si': np.float64(0.0), 'O': np.float64(0.0), 'S': np.float64(0.0)}
Mantle composition:  {'FeO': np.float64(0.0), 'SiO2': np.float64(52.55497015), 'MgO': np.float64(39.17101638), 'CaO': np.float64(3.81493827), 'Al2O3': np.float64(4.45907521)}
Mantle Fe# -0.0
Core Mass Percent =  35.180

The Upper mantle .tab already exists, please wait briefly for solution

../Solutions_Small/0.07CaMg_0.00FeMg_0.09AlMg_0.9SiMg_0.0NaMg_0.00Fe
The Lower mantle .tab already exists, please wait briefly for solution

iteration # 1
iterati

In [12]:
!python Group_X.py --FeMg=1e-6  # very little Fe


Core composition:  {'Fe': np.float64(100.0), 'Si': np.float64(0.0), 'O': np.float64(0.0), 'S': np.float64(0.0)}
Mantle composition:  {'FeO': np.float64(0.0), 'SiO2': np.float64(52.55497015), 'MgO': np.float64(39.17101638), 'CaO': np.float64(3.81493827), 'Al2O3': np.float64(4.45907521)}
Mantle Fe# -0.0
Core Mass Percent =  0.000

Your closest grid filename is:  0.07CaMg_0.00FeMg_0.09AlMg_0.9SiMg_0.0NaMg_0.00Fe_U/LM_results.txt


Core composition:  {'Fe': np.float64(100.0), 'Si': np.float64(0.0), 'O': np.float64(0.0), 'S': np.float64(0.0)}
Mantle composition:  {'FeO': np.float64(0.0), 'SiO2': np.float64(52.55497015), 'MgO': np.float64(39.17101638), 'CaO': np.float64(3.81493827), 'Al2O3': np.float64(4.45907521)}
Mantle Fe# -0.0
Core Mass Percent =  0.000

The Upper mantle .tab already exists, please wait briefly for solution

../Solutions_Small/0.07CaMg_0.00FeMg_0.09AlMg_0.9SiMg_0.0NaMg_0.00Fe
The Lower mantle .tab already exists, please wait briefly for solution

iteration # 1
iteration

In [13]:
!python Group_X.py --FeMg=1e6  # very little Mg


Core composition:  {'Fe': np.float64(100.0), 'Si': np.float64(0.0), 'O': np.float64(0.0), 'S': np.float64(0.0)}
Mantle composition:  {'FeO': np.float64(0.0), 'SiO2': np.float64(52.55497015), 'MgO': np.float64(39.17101638), 'CaO': np.float64(3.81493827), 'Al2O3': np.float64(4.45907521)}
Mantle Fe# -0.0
Core Mass Percent =  100.000

Your closest grid filename is:  0.07CaMg_0.00FeMg_0.09AlMg_0.9SiMg_0.0NaMg_0.00Fe_U/LM_results.txt


Core composition:  {'Fe': np.float64(100.0), 'Si': np.float64(0.0), 'O': np.float64(0.0), 'S': np.float64(0.0)}
Mantle composition:  {'FeO': np.float64(0.0), 'SiO2': np.float64(52.55497015), 'MgO': np.float64(39.17101638), 'CaO': np.float64(3.81493827), 'Al2O3': np.float64(4.45907521)}
Mantle Fe# -0.0
Core Mass Percent =  100.000

The Upper mantle .tab already exists, please wait briefly for solution

../Solutions_Small/0.07CaMg_0.00FeMg_0.09AlMg_0.9SiMg_0.0NaMg_0.00Fe
The Lower mantle .tab already exists, please wait briefly for solution

iteration # 1
itera

**Adjusting the amount of Fe relative to Mg**

Changing the number molar ratio of Fe/Mg of the whole planet appears to have a significant effect on the size and structure of the planet:
* *Default ratio (--FeMg = 0.9)*
  * Mass =  1.000 Earth masses
  * Radius =  0.999 Earth radii
  * Core Mass Fraction =  32.95
  * Core Radius Fraction =  53.02
  * CMB Pressure =  141.56 GPa
* *Equal ratio (--FeMg = 1)*
  * Mass =  1.000 Earth masses
  * Radius =  0.993 Earth radii
  * Core Mass Fraction =  35.31
  * Core Radius Fraction =  54.56
  * CMB Pressure =  138.47 GPa
* *High Mg, low Fe (--FeMg = 1e-6)*
  * Mass =  1.000 Earth masses
  * Radius =  1.078 Earth radii
  * Core Mass Fraction =  0.20
  * Core Radius Fraction =  11.65
  * CMB Pressure =  149.43 GPa
* *High Fe, Low Mg (--FeMg = 1e6)*
  * Mass =  1.000 Earth masses
  * Radius =  0.793 Earth radii
  * Core Mass Fraction =  100.00
  * Core Radius Fraction =  100.00
  * CMB Pressure =  0.30 GPa

The mass of the planet never changes, no matter the ratio, while the radius and core mass/radius fraction and CMB pressure does change. When the ratio of Fe to Mg gets larger (more iron than magnesium), the radius shrinks, the core gets significantly larger, and the CMB pressure goes close to 0 GPa. The reverse occurs when the ratio gets smaller (less iron than magnesium). This is likely due to the fact that this variable adjusts the relative quantity of iron, which, in this program, is assumed to be entirely in the core. Thus, a high ratio of Fe to Mg would indicate a larger core than mantle, and therefore a more dense planet (smaller radius with the same mass as a less dense planet) with a larger core mass/radius fraction and less pressure at the Core-Mantle Boundary.

### **Question 3**

Explore the effect of moving Fe from the core to the mantle (as FeO). Take note of the size of the planet, the planet structure (e.g. core to mantle ratio), and the effect on the mantle composition.

In [14]:
!python Group_X.py --mol_frac_Fe_mantle=0.01  # very little FeO


Core composition:  {'Fe': np.float64(100.0), 'Si': np.float64(0.0), 'O': np.float64(0.0), 'S': np.float64(0.0)}
Mantle composition:  {'FeO': np.float64(0.62447702), 'SiO2': np.float64(52.22677644), 'MgO': np.float64(38.92640238), 'CaO': np.float64(3.79111485), 'Al2O3': np.float64(4.43122931)}
Mantle Fe# 0.0089197224975223
Core Mass Percent =  32.458

Your closest grid filename is:  0.07CaMg_0.00FeMg_0.09AlMg_0.9SiMg_0.0NaMg_0.00Fe_U/LM_results.txt


Core composition:  {'Fe': np.float64(100.0), 'Si': np.float64(0.0), 'O': np.float64(0.0), 'S': np.float64(0.0)}
Mantle composition:  {'FeO': np.float64(0.62447702), 'SiO2': np.float64(52.22677644), 'MgO': np.float64(38.92640238), 'CaO': np.float64(3.79111485), 'Al2O3': np.float64(4.43122931)}
Mantle Fe# 0.0089197224975223
Core Mass Percent =  32.458

The Upper mantle .tab already exists, please wait briefly for solution

../Solutions_Small/0.07CaMg_0.00FeMg_0.09AlMg_0.9SiMg_0.0NaMg_0.00Fe
The Lower mantle .tab already exists, please wait b

In [15]:
!python Group_X.py --mol_frac_Fe_mantle=0.2  # small increment


Core composition:  {'Fe': np.float64(100.0), 'Si': np.float64(0.0), 'O': np.float64(0.0), 'S': np.float64(0.0)}
Mantle composition:  {'FeO': np.float64(11.16482669), 'SiO2': np.float64(46.68729882), 'MgO': np.float64(34.79764028), 'CaO': np.float64(3.38900702), 'Al2O3': np.float64(3.96122719)}
Mantle Fe# 0.15254237288135578
Core Mass Percent =  25.769

Your closest grid filename is:  0.07CaMg_0.00FeMg_0.09AlMg_0.9SiMg_0.0NaMg_0.10Fe_U/LM_results.txt


Core composition:  {'Fe': np.float64(100.0), 'Si': np.float64(0.0), 'O': np.float64(0.0), 'S': np.float64(0.0)}
Mantle composition:  {'FeO': np.float64(11.16482669), 'SiO2': np.float64(46.68729882), 'MgO': np.float64(34.79764028), 'CaO': np.float64(3.38900702), 'Al2O3': np.float64(3.96122719)}
Mantle Fe# 0.15254237288135578
Core Mass Percent =  25.769

The Upper mantle .tab already exists, please wait briefly for solution

../Solutions_Small/0.07CaMg_0.00FeMg_0.09AlMg_0.9SiMg_0.0NaMg_0.10Fe
The Lower mantle .tab already exists, please wa

In [16]:
!python Group_X.py --mol_frac_Fe_mantle=0.39  # maximum allowed amount of FeO


Core composition:  {'Fe': np.float64(100.0), 'Si': np.float64(0.0), 'O': np.float64(0.0), 'S': np.float64(0.0)}
Mantle composition:  {'FeO': np.float64(19.68364902), 'SiO2': np.float64(42.21023429), 'MgO': np.float64(31.460731), 'CaO': np.float64(3.06401921), 'Al2O3': np.float64(3.58136649)}
Mantle Fe# 0.25980754996299016
Core Mass Percent =  19.310

Your closest grid filename is:  0.07CaMg_0.00FeMg_0.09AlMg_0.9SiMg_0.0NaMg_0.20Fe_U/LM_results.txt


Core composition:  {'Fe': np.float64(100.0), 'Si': np.float64(0.0), 'O': np.float64(0.0), 'S': np.float64(0.0)}
Mantle composition:  {'FeO': np.float64(19.68364902), 'SiO2': np.float64(42.21023429), 'MgO': np.float64(31.460731), 'CaO': np.float64(3.06401921), 'Al2O3': np.float64(3.58136649)}
Mantle Fe# 0.25980754996299016
Core Mass Percent =  19.310

The Upper mantle .tab already exists, please wait briefly for solution

../Solutions_Small/0.07CaMg_0.00FeMg_0.09AlMg_0.9SiMg_0.0NaMg_0.20Fe
The Lower mantle .tab already exists, please wait b

**Moving Fe from the core to the mantle (as FeO)**

Moving iron from the core to the mantle of this planet appears to have a significant effect on the size of the planet, the planet structure (e.g. core to mantle ratio), and the mantle composition:
* *Small amount of FeO (mol_frac_Fe_mantle=0.01)*
  * Mass =  1.000 Earth masses
  * Radius =  1.000 Earth radii
  * Core Mass Fraction =  32.59
  * Core Radius Fraction =  52.79
  * CMB Pressure =  142.02 GPa
  * Mantle composition: FeO=0.624, SiO2=52.227, MgO=38.926, CaO=3.791, Al2O3=4.431
* *Intermediate amount of FeO (mol_frac_Fe_mantle=0.2)*
  * Mass =  1.000 Earth masses
  * Radius =  1.004 Earth radii
  * Core Mass Fraction =  25.92
  * Core Radius Fraction =  48.72
  * CMB Pressure =  155.48 GPa
  * Mantle composition: FeO=11.165, SiO2=46.687, MgO=34.798, CaO=3.389, Al2O3=3.961
* *Maximum allowed amount of FeO (mol_frac_Fe_mantle=0.39)*
  * Mass =  1.000 Earth masses
  * Radius =  1.005 Earth radii
  * Core Mass Fraction =  19.47
  * Core Radius Fraction =  44.26
  * CMB Pressure =  169.31 GPa
  * Mantle composition: FeO=19.684, SiO2=42.210, MgO=31.461, CaO=3.064, Al2O3=3.581

This mass of the planet never changes, despite the amount of iron moved from the core to the mantle. The radius and CMB pressure both get slightly larger and the core mass/radius fractions both get smaller (linearly) as the amount moved increases. This is because iron is being moved from the core to the mantle in the form of FeO, which decreases its density because Fe is more dense than FeO (and increases its radius), and the core gets smaller while the mantle increases in size (which would increase the pressure at the CMB).

As for the mantle composition, FeO is the only compound that increases in relative frequency, while the other four decrease. This is because FeO is the only compound that is being added/removed while the others remain constant, and the total amount of substance in the mantle is increasing, so of course the other four relative frequencies would decrease.


### **Question 4**

Explore the effect of adding light elements to the core. Take note of the size of the planet, the planet structure (e.g. core to mantle ratio), and the effect on the mantle composition.

In [17]:
!python Group_X.py --wt_frac_Si_core=0.01 --wt_frac_O_core=0.01 --wt_frac_S_core=0.01  # very little light elements


Core composition:  {'Fe': np.float64(97.0), 'Si': np.float64(1.0), 'O': np.float64(1.0), 'S': np.float64(1.0)}
Mantle composition:  {'FeO': np.float64(2.39338503), 'SiO2': np.float64(50.81249377), 'MgO': np.float64(38.63362071), 'CaO': np.float64(3.7626003), 'Al2O3': np.float64(4.39790019)}
Mantle Fe# 0.033587674234745075
Core Mass Percent =  32.317

Your closest grid filename is:  0.07CaMg_0.00FeMg_0.09AlMg_0.9SiMg_0.0NaMg_0.02Fe_U/LM_results.txt


Core composition:  {'Fe': np.float64(97.0), 'Si': np.float64(1.0), 'O': np.float64(1.0), 'S': np.float64(1.0)}
Mantle composition:  {'FeO': np.float64(2.39338503), 'SiO2': np.float64(50.81249377), 'MgO': np.float64(38.63362071), 'CaO': np.float64(3.7626003), 'Al2O3': np.float64(4.39790019)}
Mantle Fe# 0.033587674234745075
Core Mass Percent =  32.317

The Upper mantle .tab already exists, please wait briefly for solution

../Solutions_Small/0.07CaMg_0.00FeMg_0.09AlMg_0.9SiMg_0.0NaMg_0.02Fe
The Lower mantle .tab already exists, please wait b

In [18]:
!python Group_X.py --wt_frac_Si_core=0.05 --wt_frac_O_core=0.05 --wt_frac_S_core=0.05  # small increment


Core composition:  {'Fe': np.float64(85.0), 'Si': np.float64(5.0), 'O': np.float64(5.0), 'S': np.float64(5.0)}
Mantle composition:  {'FeO': np.float64(10.28035014), 'SiO2': np.float64(44.87258022), 'MgO': np.float64(37.02611854), 'CaO': np.float64(3.60604267), 'Al2O3': np.float64(4.21490843)}
Mantle Fe# 0.1347722482943478
Core Mass Percent =  30.995

Your closest grid filename is:  0.07CaMg_0.00FeMg_0.09AlMg_0.8SiMg_0.0NaMg_0.10Fe_U/LM_results.txt


Core composition:  {'Fe': np.float64(85.0), 'Si': np.float64(5.0), 'O': np.float64(5.0), 'S': np.float64(5.0)}
Mantle composition:  {'FeO': np.float64(10.28035014), 'SiO2': np.float64(44.87258022), 'MgO': np.float64(37.02611854), 'CaO': np.float64(3.60604267), 'Al2O3': np.float64(4.21490843)}
Mantle Fe# 0.1347722482943478
Core Mass Percent =  30.995

The Upper mantle .tab already exists, please wait briefly for solution

../Solutions_Small/0.07CaMg_0.00FeMg_0.09AlMg_0.8SiMg_0.0NaMg_0.10Fe
The Lower mantle .tab already exists, please wait b

In [19]:
!python Group_X.py --wt_frac_Si_core=0.11 --wt_frac_O_core=0.11 --wt_frac_S_core=0.11  # maximum allowed amounts (sum < 0.334)


Core composition:  {'Fe': np.float64(67.0), 'Si': np.float64(11.0), 'O': np.float64(11.0), 'S': np.float64(11.0)}
Mantle composition:  {'FeO': np.float64(18.9622221), 'SiO2': np.float64(37.59200523), 'MgO': np.float64(35.86919596), 'CaO': np.float64(3.49336782), 'Al2O3': np.float64(4.08320889)}
Mantle Fe# 0.2287386156494901
Core Mass Percent =  30.920

Your closest grid filename is:  0.07CaMg_0.00FeMg_0.09AlMg_0.7SiMg_0.0NaMg_0.20Fe_U/LM_results.txt


Core composition:  {'Fe': np.float64(67.0), 'Si': np.float64(11.0), 'O': np.float64(11.0), 'S': np.float64(11.0)}
Mantle composition:  {'FeO': np.float64(18.9622221), 'SiO2': np.float64(37.59200523), 'MgO': np.float64(35.86919596), 'CaO': np.float64(3.49336782), 'Al2O3': np.float64(4.08320889)}
Mantle Fe# 0.2287386156494901
Core Mass Percent =  30.920

The Upper mantle .tab already exists, please wait briefly for solution

../Solutions_Small/0.07CaMg_0.00FeMg_0.09AlMg_0.7SiMg_0.0NaMg_0.20Fe
The Lower mantle .tab already exists, please wa

**Adding light elements to core**

Adding light elements to the core (Si, O, S) appears to have a significant effect on the size of the planet, the planet structure (e.g. core to mantle ratio), and the mantle composition:
* *Small amount of light elements (wt_frac_Si_core=0.01, wt_frac_O_core=0.01, wt_frac_S_core=0.01)*
  * Mass =  1.000 Earth masses
  * Radius =  1.002 Earth radii
  * Core Mass Fraction =  32.45
  * Core Radius Fraction =  53.53
  * CMB Pressure =  138.37 GPa
  * Mantle composition: FeO=2.393, SiO2=50.812, MgO=38.634, CaO=3.763, Al2O3=4.398
* *Intermediate amount of light elements (wt_frac_Si_core=0.05, wt_frac_O_core=0.05, wt_frac_S_core=0.05)*
  * Mass =  1.000 Earth masses
  * Radius =  1.009 Earth radii
  * Core Mass Fraction =  31.13
  * Core Radius Fraction =  55.71
  * CMB Pressure =  126.84 GPa
  * Mantle composition: FeO=10.280, SiO2=44.872, MgO=37.026, CaO=3.606, Al2O3=4.215
* *Maximum allowed amount of light elements (wt_frac_Si_core=0.11, wt_frac_O_core=0.11, wt_frac_S_core=0.11)*
  * Mass =  1.000 Earth masses
  * Radius =  1.018 Earth radii
  * Core Mass Fraction =  31.06
  * Core Radius Fraction =  59.62
  * CMB Pressure =  111.12 GPa
  * Mantle composition: FeO=18.962, SiO2=37.592, MgO=35.869, CaO=3.493, Al2O3=4.083

This mass of the planet never changes, despite the amount of light elements added to the core. The radius (*barely*) and core radius fraction both get larger, while the core mass fraction and CMB pressure both get smaller as the amount of light elements added increases. This is because light elements are being added to the core, which decreases the relative frequency of Fe in the core and increases the relative frequency of FeO in the mantle. This decreases its density because these elements are less dense than Fe (and increases its radius), and the radius of the core gets larger while the mass of the core gets smaller (less dense elements). The mantle-to-core ratio becomes smaller, which decreases the pressure at the CMB.

As for the mantle composition, FeO is the only compound that increases in relative frequency, while the other four decrease. This is because FeO is the only compound that is being added/removed while the others remain constant, and the total amount of substance in the mantle is increasing, so of course the other four relative frequencies would decrease. It is also important to note that changing the amount of Si, O, and S (ie. not keeping them the same frequency) does change the core/mantle composition, as well as the planet structure, but only slightly.

### **Question 5**

Take a look at the mineral proportions through the mantle and how they change with Si/Mg ratios (output is found in .tsv file). Open this in excel or Matlab or other appropriate data viewer, and identify the most abundant minerals. Pick one and examine how it changes with Si/Mg.

In [20]:
!python Group_X.py  # no adjustment (control)


Core composition:  {'Fe': np.float64(100.0), 'Si': np.float64(0.0), 'O': np.float64(0.0), 'S': np.float64(0.0)}
Mantle composition:  {'FeO': np.float64(0.0), 'SiO2': np.float64(52.55497015), 'MgO': np.float64(39.17101638), 'CaO': np.float64(3.81493827), 'Al2O3': np.float64(4.45907521)}
Mantle Fe# -0.0
Core Mass Percent =  32.816

Your closest grid filename is:  0.07CaMg_0.00FeMg_0.09AlMg_0.9SiMg_0.0NaMg_0.00Fe_U/LM_results.txt


Core composition:  {'Fe': np.float64(100.0), 'Si': np.float64(0.0), 'O': np.float64(0.0), 'S': np.float64(0.0)}
Mantle composition:  {'FeO': np.float64(0.0), 'SiO2': np.float64(52.55497015), 'MgO': np.float64(39.17101638), 'CaO': np.float64(3.81493827), 'Al2O3': np.float64(4.45907521)}
Mantle Fe# -0.0
Core Mass Percent =  32.816

The Upper mantle .tab already exists, please wait briefly for solution

../Solutions_Small/0.07CaMg_0.00FeMg_0.09AlMg_0.9SiMg_0.0NaMg_0.00Fe
The Lower mantle .tab already exists, please wait briefly for solution

iteration # 1
iterati

In [21]:
!python Group_X.py --SiMg=1  # equal quantities


Core composition:  {'Fe': np.float64(100.0), 'Si': np.float64(0.0), 'O': np.float64(0.0), 'S': np.float64(0.0)}
Mantle composition:  {'FeO': np.float64(0.0), 'SiO2': np.float64(55.17263759), 'MgO': np.float64(37.00984808), 'CaO': np.float64(3.60445806), 'Al2O3': np.float64(4.21305627)}
Mantle Fe# -0.0
Core Mass Percent =  31.578

Your closest grid filename is:  0.07CaMg_0.00FeMg_0.09AlMg_1.0SiMg_0.0NaMg_0.00Fe_U/LM_results.txt


Core composition:  {'Fe': np.float64(100.0), 'Si': np.float64(0.0), 'O': np.float64(0.0), 'S': np.float64(0.0)}
Mantle composition:  {'FeO': np.float64(0.0), 'SiO2': np.float64(55.17263759), 'MgO': np.float64(37.00984808), 'CaO': np.float64(3.60445806), 'Al2O3': np.float64(4.21305627)}
Mantle Fe# -0.0
Core Mass Percent =  31.578

The Upper mantle .tab already exists, please wait briefly for solution

../Solutions_Small/0.07CaMg_0.00FeMg_0.09AlMg_1.0SiMg_0.0NaMg_0.00Fe
The Lower mantle .tab already exists, please wait briefly for solution

iteration # 1
iterati

In [22]:
!python Group_X.py --SiMg=0.5  # minimum Si


Core composition:  {'Fe': np.float64(100.0), 'Si': np.float64(0.0), 'O': np.float64(0.0), 'S': np.float64(0.0)}
Mantle composition:  {'FeO': np.float64(0.0), 'SiO2': np.float64(38.09545149), 'MgO': np.float64(51.10891681), 'CaO': np.float64(4.97759263), 'Al2O3': np.float64(5.81803907)}
Mantle Fe# -0.0
Core Mass Percent =  38.925

Your closest grid filename is:  0.07CaMg_0.00FeMg_0.09AlMg_0.5SiMg_0.0NaMg_0.00Fe_U/LM_results.txt


Core composition:  {'Fe': np.float64(100.0), 'Si': np.float64(0.0), 'O': np.float64(0.0), 'S': np.float64(0.0)}
Mantle composition:  {'FeO': np.float64(0.0), 'SiO2': np.float64(38.09545149), 'MgO': np.float64(51.10891681), 'CaO': np.float64(4.97759263), 'Al2O3': np.float64(5.81803907)}
Mantle Fe# -0.0
Core Mass Percent =  38.925

The Upper mantle .tab already exists, please wait briefly for solution

../Solutions_Small/0.07CaMg_0.00FeMg_0.09AlMg_0.5SiMg_0.0NaMg_0.00Fe
The Lower mantle .tab already exists, please wait briefly for solution

iteration # 1
iterati

In [23]:
!python Group_X.py --SiMg=2  # minimum Mg


Core composition:  {'Fe': np.float64(100.0), 'Si': np.float64(0.0), 'O': np.float64(0.0), 'S': np.float64(0.0)}
Mantle composition:  {'FeO': np.float64(0.0), 'SiO2': np.float64(71.111297), 'MgO': np.float64(23.85075659), 'CaO': np.float64(2.32286962), 'Al2O3': np.float64(2.71507679)}
Mantle Fe# -0.0
Core Mass Percent =  22.924

Your closest grid filename is:  0.07CaMg_0.00FeMg_0.09AlMg_2.0SiMg_0.0NaMg_0.00Fe_U/LM_results.txt


Core composition:  {'Fe': np.float64(100.0), 'Si': np.float64(0.0), 'O': np.float64(0.0), 'S': np.float64(0.0)}
Mantle composition:  {'FeO': np.float64(0.0), 'SiO2': np.float64(71.111297), 'MgO': np.float64(23.85075659), 'CaO': np.float64(2.32286962), 'Al2O3': np.float64(2.71507679)}
Mantle Fe# -0.0
Core Mass Percent =  22.924

The Upper mantle .tab already exists, please wait briefly for solution

../Solutions_Small/0.07CaMg_0.00FeMg_0.09AlMg_2.0SiMg_0.0NaMg_0.00Fe
The Lower mantle .tab already exists, please wait briefly for solution

iteration # 1
iteration #

**Changing mineral proportions through the mantle with Si/Mg ratios**

Looking at the control .tsv file, perovskite is clearly one of the most abundant minerals, ranging from depth 629.772 to 2576.943 km. Its abundance at these depths (with a default ratio of Si/Mg) remains around 78.475% for nearly the entirety of the near 2000 km where this substance can be found (ranges from 791.113 to 2560.058 km). The remaining ~500 km, it slowly tapers off until the mineral cannot be found at a depth less than 630 km.

Very little changes when we change the ratio of Si to Mg to 1 (equal quantities of each). Perovskite can be found in depths ranging from 633.158 to 2602.326 km, and remains around 87.444% abundant for nearly the exact same range of depths (about 2000 km) as the default group.

When we change the ratio of Si to Mg to 0.5 (the minimum ratio allowed - 5 Si molecules for 100 Mg), perovskite can be found in depths ranging from 634.112 to 2477.213 km, with a relative abundance of around 41.823% for around 1700 km. When we do the opposite, and change the ratio of Si to Mg to 2 (the maximum ratio allowed - 2 Si molecules for 1 Mg), perovskite can be found in depths ranging from 651.982 to 2738.246 km, with a relative abundance of around 51.001% for around 2000 km.

Thus, we can conclude that changing the Si/Mg ratio affecrs the mineral proportions throughout the mantle. While the ratios barely affected the depths at which perovskite could be found, they did affect its relative abundance. A low ratio of Si to Mg means less abundance of perovskite, while a high ratio of Si to Mg also means a lower abundance of perovskite (but still higher than a low ratio). Perovskite can be found in the greatest relative abundance when there are an equal amoung of Si and Mg atoms.

### **Question 6**

What parameters are most useful in better reproducing the sizes of each of these planets? Is it realistic?

When it comes to strictly reproducing the sizes of these planets, it is clear that, outside of mass, the FeMg and SiMg parameters have the greatest impact on the predicted size of the planet. It is clear from the investigations from above, that mass, FeMg, and SiMg affect the planet's radius most significantly. It is important to note that wt_frac_Si_core, wt_frac_O_core, and wt_frac_S_core do impact the size of the planet rather significantly, but less than the other three parameters do. Also, mol_frac_Fe_mantle does not have much of an impact on the radius of the planet, as shown in the exercises above. I do not think the parameters CaMg and AlMg have as much impact (as they take up such a small proportion of relative abundance and have very strict boundaries within the program). These parameters are realistic, as we can determine the mass of the planet easily, along with its relative FeMg and SiMg ratios, as Fe, Si, and Mg are all very abundant elements within these simulated planets.

$ $

## **Model Exoplanet System**

The planet we have selected is TRAPPIST-1e from the TRAPPIST-1 system. We can confirm this planet is likely rocky after consult some of the papers we previously read this semester (Fulton et al., 2017; Seager et al., 2007; Chen and Kipping, 2016). The planet is is smaller than Earth, but has around the same density, which indicates a rocky exterior.

We found the radius and mass of this planet using the [NASA Exoplanet Archive](https://exoplanetarchive.ipac.caltech.edu/cgi-bin/TblView/nph-tblView?app=ExoTbls&config=PS). The [$\frac{Fe}{H}$] value was found via [IOPscience.org](https://iopscience.iop.org/article/10.3847/1538-4357/aaa023/pdf). By referencing Figure 3 in Griffith et al., 2020, we estimated: [$\frac{Mg}{Fe}$] ≈ 0.08 and [$\frac{Si}{Fe}$] ≈ 0.07. We can find our [$\frac{X}{H}$] values using the following general formula: [$\frac{X}{H}$] = [$\frac{X}{Fe}$] + [$\frac{Fe}{H}$]. Knowing [$\frac{Fe}{H}$] is estimated to be 0.04 for our TRAPPIST-1 star, we can further estimate: [$\frac{Mg}{H}$] ≈ 0.12 and [$\frac{Si}{H}$] ≈ 0.11.

To find [$\frac{Fe}{Mg}$] and [$\frac{Si}{Mg}$], we use this formula: [$\frac{X}{Mg}$] = $10^{[\frac{X}{H}] - [\frac{Mg}{H}]}$.

In [240]:
# Estimated ratios for [Fe/H], [Mg/Fe], and [Si/Fe]
Fe_H = 0.04
Mg_H = 0.12
Si_H = 0.11

# Find [Fe/Mg] and [Si/Mg]
Fe_Mg = 10**((Fe_H - Mg_H))
Si_Mg = 10**((Si_H - Mg_H))
print(f'[Fe/Mg]=',round(Fe_Mg,3),'and [Si/Mg]=',round(Si_Mg,3))

[Fe/Mg]= 0.832 and [Si/Mg]= 0.977


We have determined our planet mass and molar ratios [Fe/Mg] and [Si/Mg].

This planet's radius is about 0.7 times that of Earth, which puts it in the range of likely rocky planets. In general, planets smaller than around 1.6 Earth radii are expected to have solid, rocky compositions without thick gas envelopes.

The luminosity of TRAPPIST-1 is about 0.000522 times the Sun’s. Thus, the planet ends up receiving over four times the solar flux Earth gets, which indicated that it has a high surface temperature and atmosphere retention.

The likely refractory composition of the star indicates that TRAPPIST-1e likely has a solid, rocky makeup. Using our estimated molar ratios of [Fe/Mg] and [Si/Mg] and known mass of 0.772 Earth masses for our planet, we have a known radius of 0.910 Earth radii and density of 1.024 Earth density. We will continue our analysis, but it is important to note there will be varied results given the uncertainties in the measured mass of (+0.026, -0.027) Earth masses. Small variations in this value create large discrepencies when trying to replicate this planet using ExoPlex.

We will now create a function that will try and determine the values for mol_frac_Fe_mantle, wt_frac_Si_core, wt_frac_O_core, and wt_frac_S_core create a planet that reflects our desired radius and density.

In [78]:
## Prepare for analysis

# Function to calculate density
def calculate_earth_density(mass_earth_masses, radius_earth_radii):
    volume = radius_earth_radii ** 3  # Volume scales with cube of radius
    density = mass_earth_masses / volume  # Density = mass / volume
    return density

# Mass and radius of TRAPPIST-1 e
radius = 0.910                      # in R_earth
mass = 0.772                        # in M_earth
radius_kg = radius * const.R_earth  # in kg
mass_kg = mass * const.M_earth      # in kg
density = calculate_earth_density(mass, radius)  # in Earth density

*Our most realistic parameters:*

In [274]:
!python Group_X.py --mass=0.772 --FeMg=1.361 --SiMg=0.977 --mol_frac_Fe_mantle=0.15 --wt_frac_Si_core=0.015 --wt_frac_O_core=0.01 --wt_frac_S_core=0.025


Core composition:  {'Fe': np.float64(95.0), 'Si': np.float64(1.5), 'O': np.float64(1.0), 'S': np.float64(2.5)}
Mantle composition:  {'FeO': np.float64(5.04424717), 'SiO2': np.float64(50.84950794), 'MgO': np.float64(36.41448739), 'CaO': np.float64(3.54647477), 'Al2O3': np.float64(4.14528274)}
Mantle Fe# 0.0721089940105226
Core Mass Percent =  40.531

Your closest grid filename is:  0.07CaMg_0.00FeMg_0.09AlMg_0.9SiMg_0.0NaMg_0.06Fe_U/LM_results.txt


Core composition:  {'Fe': np.float64(95.0), 'Si': np.float64(1.5), 'O': np.float64(1.0), 'S': np.float64(2.5)}
Mantle composition:  {'FeO': np.float64(5.04424717), 'SiO2': np.float64(50.84950794), 'MgO': np.float64(36.41448739), 'CaO': np.float64(3.54647477), 'Al2O3': np.float64(4.14528274)}
Mantle Fe# 0.0721089940105226
Core Mass Percent =  40.531

The Upper mantle .tab already exists, please wait briefly for solution

../Solutions_Small/0.07CaMg_0.00FeMg_0.09AlMg_0.9SiMg_0.0NaMg_0.06Fe
The Lower mantle .tab already exists, please wait bri

### **Results**

| Radius (R🜨) | Mass (M🜨) | Density (ρ/ρ🜨) | [Fe/Mg] | [Si/Mg] | FeO (mantle) | Si (core) | O (core) | S (core) | Mantle Composition (%) |
|-------------|-----------|----------------|--------|--------|--------------|-----------|----------|----------|----------------------|
| 0.910        | 0.772      | 1.024           | 1.300    | 0.977    | 0         | 0      | 0     | 0     | SiO₂=54.6<br>MgO=37.5<br>CaO=3.7<br>Al₂O₃=4.3           |
| 0.935        | 0.772      | 0.944           | 0.832    | 0.977    | 0.15         | 0.03      | 0     | 0.01     | SiO₂=50.3<br>MgO=36.2<br>CaO=3.5<br>Al₂O₃=4.1           |
| 0.934        | 0.772      | 0.947           | 0.832    | 0.977    | 0         | 0      | 0     | 0     | SiO₂=54.6<br>MgO=37.5<br>CaO=3.7<br>Al₂O₃=4.3           |
| 0.935        | 0.772      | 0.944           | 0.832    | 0.977    | 0.15         | 0.05      | 0     | 0.03     | SiO₂=47.8<br>MgO=35.5<br>CaO=3.5<br>Al₂O₃=4.1           |
| 0.923        | 0.772      | 0.982           | 1.023    | 0.977    | 0.12         | 0.005      | 0     | 0.035     | SiO₂=53.6<br>MgO=37.2<br>CaO=3.6<br>Al₂O₃=4.2           |
| 0.910        | 0.772      | 1.024           | 1.160    | 0.670    | 0.2         | 0.03      | 0     | 0.05     | SiO₂=53.6<br>MgO=37.2<br>CaO=3.6<br>Al₂O₃=4.2           |
| 0.910        | 0.772      | 1.024           | 1.361    | 0.977    | 0.15         | 0.015     | 0.01     | 0.025     | SiO₂=50.8<br>MgO=36.4<br>CaO=3.5<br>Al₂O₃=4.1          |


There is no combination of parameters that would decrease our estimated radius to 0.91 Earth radii, our known radius, without increasing our [Fe/Mg] value, which indicates that TRAPPIST-1e has a very iron-abundant core.

We can also show a few iterations using our estimated [Fe/Mg] value of 0.832 that would give us radii close to 0.91 Earth radii; these estimated radii will fall within the known uncertainty bounds of our exoplanet: (+0.026, -0.027). As we can see from the rows 2-4, we cannot estimate a radius of less than 0.935 Earth radii and a [Fe/Mg] ratio of 0.832, no matter the combination of parameters.

We can do a combination of parameter adjustments and increase our [Fe/Mg] ratio within a reasonable amount (lower end of the spread of probable [X/Fe] ratios from Figure 3 in Griffiths et al. 2020) and then adjust our parameters as indicated in the chart and obtain an estimated radius of 0.923 Earth radii and density of 0.982 Earth densities.

Many studies [like this one](https://www.aanda.org/articles/aa/pdf/2018/05/aa32233-17.pdf) indicate that TRAPPIST-1e cannot accommodate a Hydrogen-dominated atmosphere, so the issue is not due to a lack of atmosphere (and if there was an atmosphere, it would likely be a thick, Earth-like atmosphere). Thus, our issue is the fact that our core is too light. To fix this, we can increase our [Fe/Mg] ratio, decrease our [Si/Mg] ratio, increase FeO in the mantle, increase Si and S in the core and decrease O, or some combination of all of the above. Thus, we have a dense planet, similar to Earths, with a lot of heavy materials within its core. We tried to increase the abundance of Fe within the core and mantle as much as possible.

The parameters of our most realistic simulated planet is as follows:
* Mass = 0.772
* [Fe/Mg] = 1.361
* [Si/Mg] = 0.977
* mol_frac_Fe_mantle = 0.15
* wt_frac_Si_core = 0.015
* wt_frac_O_core = 0.005
* wt_frac_S_core = 0.025

which gave us an estimated radius of 0.910 Earth radii and a density of 1.024 Earth densities, which is equal to our known values.

*Mantle minerology of Earth:*

In [254]:
!python Group_X.py


Core composition:  {'Fe': np.float64(100.0), 'Si': np.float64(0.0), 'O': np.float64(0.0), 'S': np.float64(0.0)}
Mantle composition:  {'FeO': np.float64(0.0), 'SiO2': np.float64(52.55497015), 'MgO': np.float64(39.17101638), 'CaO': np.float64(3.81493827), 'Al2O3': np.float64(4.45907521)}
Mantle Fe# -0.0
Core Mass Percent =  32.816

Your closest grid filename is:  0.07CaMg_0.00FeMg_0.09AlMg_0.9SiMg_0.0NaMg_0.00Fe_U/LM_results.txt


Core composition:  {'Fe': np.float64(100.0), 'Si': np.float64(0.0), 'O': np.float64(0.0), 'S': np.float64(0.0)}
Mantle composition:  {'FeO': np.float64(0.0), 'SiO2': np.float64(52.55497015), 'MgO': np.float64(39.17101638), 'CaO': np.float64(3.81493827), 'Al2O3': np.float64(4.45907521)}
Mantle Fe# -0.0
Core Mass Percent =  32.816

The Upper mantle .tab already exists, please wait briefly for solution

../Solutions_Small/0.07CaMg_0.00FeMg_0.09AlMg_0.9SiMg_0.0NaMg_0.00Fe
The Lower mantle .tab already exists, please wait briefly for solution

iteration # 1
iterati

| Mantle Minerology       | TRAPPIST-1e (%) | Earth (%) |
|-------------------------|-----------------|-----------|
| **Fe**                  | 0.07            | 0         |
| **FeO**                 | 5.04            | 0         |
| **SiO₂**                | 50.8            | 52.5      |
| **MgO**                 | 36.4            | 39.2      |
| **CaO**                 | 3.5             | 3.8       |
| **Al₂O₃**               | 4.1             | 4.5       |

Here, we can see the differences between the mantle minerology of TRAPPIST-1e and Earth. For the most part, these values are quite similar. This is because they have nearly identical densities. The main difference is iron, which can be found in the mantle of TRAPPIST-1e, but is solely within the core of Earth, according to ExoPlex. That being said, there are hundreds of combinations of parameters that would lead to similar results, and there is a lot of uncertainty when it comes to the 'known' mass and radius estimates. We can, however, conclude that TRAPPIST-1e is similar in terms of composition to the Earth, and has a very heavy core (high abundance of iron and other heavy elements).